src/openrct2/scripting/bindings/entity/ScStaff.cpp

```diff
@@ -145,6 +161,34 @@ namespace OpenRCT2::Scripting
         }
     }
 
+    std::string ScStaff::action_get() const
+    {
+        auto peep = GetStaff();
+        if (peep != nullptr)
+        {
+            return std::string(StaffActionMap[peep->Action]);
+        }
+        return {};
+    }
+
+    void ScStaff::action_set(const std::string value)
+    {
+        ThrowIfGameStateNotMutable();
+        auto peep = GetStaff();
+        if (peep != nullptr)
+        {
+            if (!StaffActionMap.TryGet(value))
+            {
+                return;
```

Comment:
I think it would be good if this throws an error rather than being a silent failure.

Reply:
Is this a valid solution?

https://github.com/OpenRCT2/OpenRCT2/pull/21020/commits/0f64670e6540cb66ad9d702a6baaf9e442806a02#diff-34aab9baf955b169e0245bcfc39a571b29cf929eed6216b5d444899667e25aacR488

Reply:
No, use duk_error, instead of printf. You can look for existing uses of duk_error.